modules/claims_api/spec/lib/claims_api/bd_spec.rb

```diff
@@ -112,29 +112,63 @@
 
       context 'when the doctype is L075' do
-        let(:pdf_path) { 'modules/claims_api/spec/fixtures/21-22A/signed_filled_final.pdf' }
-        let(:json_body) do
-          res = subject.send(:generate_upload_body, claim: power_of_attorney, pdf_path:,
-                                                    doc_type: 'L075')
-          temp_io = res[:parameters].instance_variable_get(:@io).path
-          temp_io_contents = File.read(temp_io)
-          JSON.parse(temp_io_contents)
-        end
-
-        it 'the systemName is Lighthouse' do
-          expect(json_body['data']['systemName']).to eq('Lighthouse')
-        end
-
-        it 'the docType is L075' do
-          expect(json_body['data']['docType']).to eq('L075')
+        context 'when the api version is v2' do
+          let(:pdf_path) { 'modules/claims_api/spec/fixtures/21-22A/signed_filled_final.pdf' }
+          let(:json_body) do
+            res = subject.send(:generate_upload_body, claim: power_of_attorney, pdf_path:, action: 'post',
+                                                      doc_type: 'L075')
+            temp_io = res[:parameters].instance_variable_get(:@io).path
+            temp_io_contents = File.read(temp_io)
+            JSON.parse(temp_io_contents)
+          end
+
+          it 'the systemName is Lighthouse' do
+            expect(json_body['data']['systemName']).to eq('Lighthouse')
+          end
+
+          it 'the docType is L075' do
+            expect(json_body['data']['docType']).to eq('L075')
+          end
+
+          it 'the fileName ends in 21-22a.pdf' do
+            expect(json_body['data']['fileName']).to end_with('21-22a.pdf')
+          end
+
+          it 'the claimId is not present' do
+            expect(json_body['data']).not_to have_key('claimId')
+          end
         end
 
-        it 'the fileName ends in 21-22a.pdf' do
-          expect(json_body['data']['fileName']).to end_with('21-22a.pdf')
-        end
-
-        it 'the claimId is not present' do
-          expect(json_body['data']).not_to have_key('claimId')
+        context 'when the api version is v1' do
+          context 'the doc type is 21-22a' do
+            let(:pdf_path) { 'modules/claims_api/spec/fixtures/21-22A/signed_filled_final.pdf' }
+            let(:json_body) do
+              res = subject.send(:generate_upload_body, claim: power_of_attorney, pdf_path:, action: 'put',
+                                                        doc_type: 'L075')
+              temp_io = res[:parameters].instance_variable_get(:@io).path
+              temp_io_contents = File.read(temp_io)
+              JSON.parse(temp_io_contents)
+            end
+
+            it 'the fileName ends in representative.pdf' do
+              expect(json_body['data']['fileName']).to end_with('_representative.pdf')
+            end
+          end
+
+          context 'the doc type is 21-22' do
+            let(:pdf_path) { 'modules/claims_api/spec/fixtures/21-22A/signed_filled_final.pdf' }
```

Comment:
Would this be

```suggestion
          context 'the doc type is 21-22' do
            let(:pdf_path) { 'modules/claims_api/spec/fixtures/21-22/signed_filled_final.pdf' }
```

for completenessʼ sake?

Reply:
Yep, nice catch, committed it